### PR TITLE
OSX CI: bump to XCode 11

### LINKED ISSUE #599: OSX Circle CI: switch to latest XCode 11 (beta) images. Use c++17
XCode 11 is now avail on CircleCI, that would finally allow us to use c++17 with filesystem. 
- [ ] switch to XCode 11
- [ ] test working c++17 in CircleCI
- [ ] fully switch repo to c++17 (drop c++11) 
- [ ] possible related cleanups (rm boost?, ...) 

Related #55

### PR DESCRIPTION
**Pull request by breznak**
which is still beta,
but offers c++17

- [x] OSX CI runs on XCode11
- [ ] c++17  (boost OFF) -> not yet, needs macOS 10.15, but CI is 10.14
- [x] python3

Fixes #599

### CODE REVIEW

**Comment by breznak** [comment]
Fair note, after #614 , with this PR all of our CI would be using python3, and none testing py2 even though it would be still kept supported. 
What do you think about that? For me it's an ok plan for planned obsoletion of py2.

**Comment by dkeeney** [comment]
> plan for planned obsoletion of py2.

I think that is Ok provided that we note that in the README.md.

**Comment by breznak** [comment]
Please review, 

>  with this PR all of our CI would be using python3

one more point for supporting the change to py3 in all CI: as we now want to work on PYPI #19 , we need to have all binaries on the same python, and it should not be py2 at this time.

File: .circleci/config.yml [hunk 1]
```diff
@@ -74,9 +74,9 @@ jobs:
       - run:
           name: Restoring system python
           command: |
+            echo 'default to python3' && rm /usr/local/bin/python && ln -s /usr/local/bin/python3 /usr/local/bin/python
```
Comment (dkeeney) [inline .circleci/config.yml]: Also in bindings/py/setup.py,  we need to remove line 354
```
      "Programming Language :: Python :: 2",
```

**Comment by breznak** [comment]
The CircleCI was indeed wasteful, it did not have auto-cancel enabled. I had to kill a ton of pending jobs from yesterday merge-fest and enabled an experimental auto-cancel feature. Jobs seem to run, hope nothing breaks. Recent PRs may need to restart the CI builds.

**Comment by breznak** [comment]
@dkeeney one more approval please, when you have time

**Pull request merged**